undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java

```diff
@@ -982,6 +983,10 @@ private DeploymentInfo createServletConfig() throws StartException {
                 webSocketDeploymentInfo.setBuffers(servletContainer.getWebsocketsBufferPool().getValue());
                 webSocketDeploymentInfo.setWorker(servletContainer.getWebsocketsWorker().getValue());
                 webSocketDeploymentInfo.setDispatchToWorkerThread(servletContainer.isDispatchWebsocketInvocationToWorker());
+
+                // Enables compression. This fixes WFLY-6787
+                webSocketDeploymentInfo.addExtension(new PerMessageDeflateHandshake(false));
```

Comment:
I agree, but the way it is now is a regression. I opened a JIRA to have this properly configured, https://issues.jboss.org/browse/WFLY-6784.